### ISSUE TITLE
Zoom Option not working as expected

### ISSUE THREAD

**Issue by brandonhall** [watusi/jquery-mobile-iscrollview#12]
Thanks so much for your contribution making iScroll work with jQM. 

I'm setting several options before init and they are all working but the zoom. If I get into the iscroll and set zoom:true then it behaves as expected, otherwise the functionality isn't set. Any idea what I'm doing wrong?

```
$.mobile.iscrollview.prototype.options.hScroll = true;
$.mobile.iscrollview.prototype.options.hScrollbar = true;
$.mobile.iscrollview.prototype.options.resizeWrapper = false;
$.mobile.iscrollview.prototype.options.zoom = true; // not working

this._scroller = this.$el.find(":jqmData(iscroll)").iscrollview();

```

Forgot to mention that even if I set the zoom:null option to true within jquery.mobile.iscrollview.js it still doesn't work. It only works as expected by setting the option in iscroll-watusi.js

**Comment by jtara** [comment]
I see the problem. There is an option naming conflict.

 iScroll4 has a "zoom" boolean option, and an onZoom callback. iscroll4's "onWhatever" naming convention for callbacks doesn't fit into the standard naming convention for jQuery widget callbacks, I just drop the "on" and lowercase for the widget callback. Widgets want to have an option for each callback (in addition to being able to hook events).  

Not sure how I will resolve the conflict.

In the mean time, rather than altering iscroll-watusi.js, you can change the option in the iScroll4 instance. this._scroller.iscroll.options.zoom = true;

**Comment by brandonhall** [comment]
Yep, that fixed it. Thanks for the direction. I had tried that before but couldn't get it working for some reason. Who knows. Anyway, the working code is below:

```
$.mobile.iscrollview.prototype.options.hScroll = true;
$.mobile.iscrollview.prototype.options.hScrollbar = true;
$.mobile.iscrollview.prototype.options.resizeWrapper = false;
this._scroller = this.$el.find(":jqmData(iscroll)").iscrollview();
this._scroller.jqmData('iscrollview').iscroll.options.zoom = true;
```

**Comment by jtara** [comment]
Will change the callback options to include "on", which will resolve the conflict. While that doesn't go with the jQuery convention on event naming, it solves the conflict and clears up any confusion due to the events being named differently than in iScroll.

I'm changing the event prefix from iscrollview to iscroll_ anyway. They'll still be all lower case, because jQuery doesn't give me a choice. So, onZoom in iScroll will correspond to the onzoom option and iscroll_onzoom event in jquery.mobile.iscrollview

**Comment by jtara** [comment]
Fixed in 1.2. Widget events all now have a prefix of iscroll_on, externally in jQuery Mobile, and on_ internally. So, they can no longer conflict with internal iScroll options.

**Comment by brandonhall** [comment]
Just read about the update for 1.2. We will start testing the new version soon and I want to commend you on your work. Great job! It's very detailed and thorough.
